### PR TITLE
Refactor test cases

### PR DESCRIPTION
**Pull request by rtsao**
Overview of changes:
- ~~DRY up benchmarks by extracting out the shared code between the two button iteration benchmarks for each case~~ (this has been reverted)
- Generate the source style objects/stylesheets before the timed loops. This yields a more accurate benchmarking of the CSS-in-JS libraries as now only the actual conversion from these source styles to CSS is timed.
- Optimized benchmarks by eliminating redundant object literal instantiation for glamor and cxs where there's no need for an actual "stylesheet" object. Before, this would make glamor and cxs perform worse than they would in a real-world use case, where styles would be colocated in each component instead of in some separate "stylesheet" object.

### CODE REVIEW

**Comment by PepijnSenders** [comment]
Hey @rtsao, thanks for your contribution again. I don't like the way you're DRYing up the app, I did this on purpose because I wanted every case to be editable by itself without interdependencies.

**Comment by PepijnSenders** [comment]
I will check out the stuff you did with CXS and glamor though, makes sense to not generate a stylesheet object for that.

**Comment by rtsao** [comment]
I've reverted the second commit, so there's no longer any code sharing between the two button class/style overload benchmarks.

**Comment by rtsao** [comment]
Let me know if there's any further changes you'd like 😄

**Comment by PepijnSenders** [comment]
Hey @rtsao, thanks this was just regarding a personal taste for benchmark tests, that you can go into one test and see instantly what's happening. There's also a blog post that goes along with this (not yet published though). I will check out your other changes first thing tomorrow. Thanks a lot for contributing.

**Comment by rtsao** [comment]
No worries, thanks for making the repo 👍

**Comment by rtsao** [comment]
I've just squashed the commits and cleaned up some diff noise so the changes should be more clear.

File: src/classes-overload-test/cases/aphrodite.js [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-    const useStyles = StyleSheet.create(generateStyles());
+    const useStyles = StyleSheet.create(stylesheet);
 
-    const { html, css } = StyleSheetServer.renderStatic(() => {
-        const classNames = {};
-
-        for (const className in useStyles) {
-            classNames[className] = aphroditeCss(useStyles[className]);
-        }
+    const getButtonClassName = i => aphroditeCss(useStyles[buttonClassNames[i]]);
```
Comment (PepijnSenders) [inline src/classes-overload-test/cases/aphrodite.js]: Doesn't this need to happen inside the StyleSheetServer call? I think the css method from aphrodite needs a `document`.

**Comment by PepijnSenders** [comment]
LGTM 👍

**Comment by rtsao** [comment]
Thanks! I suppose it's probably worth re-running the benchmarks for the README as the timings will now be more accurate.

**Comment by PepijnSenders** [comment]
Was about to do that 😄. I saw you also have your own CSS-in-JS lib, care to contribute that one as well?

**Pull request merged**